compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -1518,31 +1515,25 @@ fn generic_simd_intrinsic<'ll, 'tcx>(
         assert_eq!(underlying_ty, non_ptr(element_ty0));
 
         // The element type of the third argument must be a signed integer type of any width:
-        match element_ty2.kind() {
-            ty::Int(_) => (),
+        let mask_elem_bitwidth = match element_ty2.kind() {
```

Comment:
This changes the `simd_scatter` intrinsic, but none of the tests tests that were touched in this test involve it? Do we have no `simd_scatter` tests? Please add one or adjust an existing one if there is one.